src/compositedataframe.jl

```diff
@@ -62,8 +62,8 @@ df = CompositeDataFrame(:MyDF, x = 1:3, y = [2, 1, 2])
 """
 function CompositeDataFrame(columns::Vector{Any},
                             cnames::Vector{Symbol} = gennames(length(columns)),
-                            typename::Symbol = symbol("CompositeDF" * string(gensym())))
-    rowtypename = symbol(string(typename, "Row"))
+                            typename::Symbol = @compat(Symbol("CompositeDF", gensym())))
```

Comment:
The `Symbol` constructor on 0.4 doesn't take multiple string arguments like this, which is why I used `@compat`.